src/convert/streams.ts

```diff
@@ -183,19 +183,17 @@ export class StandardWriter extends ComponentWriter {
 }
 
 export class ZipWriter extends ComponentWriter {
-  // compression-/speed+ (0)<---(3)---------->(9) compression+/speed-
-  // 3 appears to be a decent balance of compression and speed. It felt like
-  // higher values = diminishing returns on compression and made conversion slower
-  private zip: Archiver = createArchive('zip', { zlib: { level: 3 } });
-  private buffers: Buffer[] = [];
+  private zip = JSZip();
+  private zipBuffer?: Buffer;
 
   public constructor(rootDestination?: SourcePath) {
     super(rootDestination);
-    void pipeline(this.zip, this.getOutputStream());
+    const destination = rootDestination ? `for: ${rootDestination}` : 'in memory';
+    this.logger.debug(`generating zip ${destination}`);
```

Comment:
if the logger is only used in ZipWriter, why not create it on the lowest level so all the other writers don't suffer the perf impact of unused loggers.

then, if someone else needs it, they could hoist it higher in the hierarchy

Reply:
It's also used in `StandardWriter`